### ISSUE TITLE
[ECS] error event OnPlayerDeath

### ISSUE THREAD

**Issue by A1dyyyy** [ikkentim/SampSharp#353]
```[Event]
public void OnPlayerDeath(Player player, Player killer, Weapon reason)
{
    if (killer == null)
    {
        // If the player killed itself, remove their money.
        player.ResetMoney();
    }
    else
    {
        // If the player was killed, give their money to the killer.
        var money = player.Money;
        if (money > 0)
        {
            killer.GiveMoney(money);
            player.ResetMoney();
        }
    }
}
```
![](https://i.imgur.com/WN5mdf4.png)
![](https://i.imgur.com/R7efWoc.png)

im get this error:
```[SampSharp:ERROR] Unhandled exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> SampSharp.Entities.EntityNotFoundException: The specified entity could not be found. (Parameter 'entity')
   at SampSharp.Entities.EntityManager.GetComponent[T](EntityId entity)
   at lambda_method(Closure , Object , Object[] , IServiceProvider , IEntityManager )
   at SampSharp.Entities.EventService.<>c__DisplayClass11_0.<CreateInvoker>b__0(Object instance, EventContext eventContext)
   at SampSharp.Entities.EventService.Invoke(EventContext context)
   at SampSharp.Entities.SAMP.EntityMiddleware.Invoke(EventContext context, IEntityManager entityManager)
   at lambda_method(Closure , Object , EventContext , IServiceProvider )
   at SampSharp.Entities.EcsBuilderUseMiddlewareExtensions.<>c__DisplayClass3_1.<UseMiddleware>b__2(EventContext context)
   at SampSharp.Entities.SAMP.EntityMiddleware.Invoke(EventContext context, IEntityManager entityManager)
   at lambda_method(Closure , Object , EventContext , IServiceProvider )
   at SampSharp.Entities.EcsBuilderUseMiddlewareExtensions.<>c__DisplayClass3_1.<UseMiddleware>b__2(EventContext context)
   at SampSharp.Entities.EventService.<>c__DisplayClass12_0.<BuildInvoke>b__0(Object[] args)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at SampSharp.Core.Callbacks.Callback.Invoke(Byte[] buffer, Int32 startIndex)
   at SampSharp.Core.HostedGameModeClient.PublicCall(String name, IntPtr data, Int32 length)
```

**Comment by Hy4e** [comment]
Is your **Player** class a **[Pooled Type]**?

**Comment by A1dyyyy** [comment]
@Hy4e The problem is with SampSharp Entities

**Comment by ikkentim** [comment]
Interesting, I'll take a look at this

**Comment by TheFuseGamer** [comment]
This seems related to #350

**Comment by ikkentim** [comment]
closed in 0b6b2e5
